language/java/mistakes.go

```diff
@@ -0,0 +1,55 @@
+package java
+
+import (
+	"context"
+	"regexp"
+	"sort"
+
+	"golang.org/x/exp/maps"
+
+	pkgerrors "github.com/pkg/errors"
+	"github.com/symflower/eval-dev-quality/log"
+	"github.com/symflower/eval-dev-quality/util"
+)
+
+// Mistakes builds a Java repository and returns the list of mistakes found.
+func Mistakes(logger *log.Logger, repositoryPath string) (mistakes []string, err error) {
```

Comment:
@ahumenberger Maybe we should make this function part of the `language/language.go` interface. 

The reason is that we can simply call `ctx.Language.Mistakes()`  when defining the code repair task. Otherwise we would need to check the language first and call the right `Mistakes` function for each language.

Reply:
Yes. It is language dependent.